src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,9 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
-        return Optional.empty();
+      if ((!Properties.getControlAllCanalsBetweenTerritoriesToPass(data) && !failureMessage.isPresent())
```

Comment:
As a comment, I find this if statement to be really interesting evidence of the effect of context. Reading the comment first, then this if statement makes perfect sense.

But notice, it actually leads with the negative conditional case. That is not how you would naturally order these, but they are ordered in this way because it makes sense to check the "any" case, before we check the "all" case. 

As soon as we forget that `!Properties.getControlAllCanalsBetweenTerritoriesToPass(data) && !failureMessage.isPresent()` means "can pass through any", this code gets to a be a *lot* more challenging.